src/LibVLCSharp.WPF/ForegroundWindow.cs

```diff
@@ -109,35 +102,90 @@ void Background_Loaded(object? sender, RoutedEventArgs e)
-            var targetPoints = source.CompositionTarget.TransformFromDevice.Transform(locationFromScreen);
-            Left = targetPoints.X;
-            Top = targetPoints.Y;
+
+            if (source == null)
+            {
+                return;
+            }
+
+            /*
+             * Note that _bckgnd.ActualWidth and _bckgnd.ActualWidth are in local coordinates
+             * and not in screen coordinates.
+             *
+             * Sizes in local and screen coordinates differ when a scale transformation has
+             * been applied to _bckgnd or one of its ancestors up to the parent window.
+             * This is also the case when the video or one of its ancestors is contained
+             * in a Viewbox.
+             *
+             * To correctly position and size the ForegroundWindow, we transform the extent
+             * of _bckgnd (top-left and bottom-right points) from _bckgnd coordinates
+             * to screen coordinates and use them as the extents of ForegroundWindow.
+             *
+             * In case a scaling is detected, we should also apply that scaling to the
+             * content of ForegroundWindow, so it naturally scales up and down with the
+             * rest of the parent controls.
+             *
+             * The video view itself natively supports scaling.
+             */
+
+            var startLocationFromScreen = _bckgnd.PointToScreen(_zeroPoint);
+            var startLocationPoint = source.CompositionTarget.TransformFromDevice.Transform(startLocationFromScreen);
+            var endLocationFromScreen = _bckgnd.PointToScreen(new Point(_bckgnd.ActualWidth, _bckgnd.ActualHeight));
+            var endLocationPoint = source.CompositionTarget.TransformFromDevice.Transform(endLocationFromScreen);
+
+            Left = Math.Min(startLocationPoint.X, endLocationPoint.X);
+            Top = Math.Min(startLocationPoint.Y, endLocationPoint.Y);
+            Width = Math.Abs(endLocationPoint.X - startLocationPoint.X);
+            Height = Math.Abs(endLocationPoint.Y - startLocationPoint.Y);
+
+            if (Math.Abs(Width - _bckgnd.ActualWidth) + Math.Abs(Height - _bckgnd.ActualHeight) > 0.5)
+            {
+                ScaleWindowContent(Width / _bckgnd.ActualWidth, Height / _bckgnd.ActualHeight);
+            }
         }
 
-        void Wndhost_SizeChanged(object? sender, SizeChangedEventArgs e)
+        void ScaleWindowContent(double scaleX, double scaleY)
         {
-            var source = PresentationSource.FromVisual(_wndhost);
-            if (source == null)
+            if (VisualChildrenCount <= 0)
+            {
+                return;
+            }
+
+            var child = (FrameworkElement)GetVisualChild(0)!;
+
+            // Do not re-create and apply the ScaleTransform if already scaled
+            // That would lead to an infinite layout update cycle
+            if (child.LayoutTransform is ScaleTransform scaleTransform && 
```

Comment:
we might want to document that we now support ScaleTransform, and no other kinds of [Transforms](https://learn.microsoft.com/en-us/dotnet/api/system.windows.media.transform), I guess.

Reply:
Should I document that in [src/LibVLCSharp.WPF/README.md](https://github.com/videolan/libvlcsharp/blob/3.x/src/LibVLCSharp.WPF/README.md)?

Reply:
Yes please!

Reply:
Done!